### PR TITLE
LPS-165522 Caching is not cleared after updating Collection Contributor module

### PR DESCRIPTION
**Pull request by jkappler**
## Motivation

[Link to ticket](https://issues.liferay.com/browse/LPS-165522)

This PR solves an issue with clearing the cache for fragment entry links after a propagation of changes

## Proposed solution

As @mbowerman mentioned, there's a mismatch on how we cache the fragments in pages and how we clear the cache. The idea would be to use the segmentsExperienceId for the cache key, instead of the array of segments. The later one also creates an issue, since we cannot guarantee that it will be ordered

## Change summary:

* Change FragmentEntryFragmentRenderer so that we catch the fragmentEntryLink by segmentsExperienceId
* Added integration test for a fragment that is cacheable, so that we check the portalCache has the correct HTML for it
* Added integration test for a fragment that is not cacheable, so that we check the portalCache is null for it
* Added integration test for a fragment that is cacheable, so that we check the portalCache has the correct updated HTML for it after propagation

## Steps to reproduce

* Create a new fragment with basic HTML (make sure to make it cacheable)
* Add the fragment to a page and publish
* Visit the page so that the fragment is cached
* Edit the HTML of the fragment, changing the text and save
* Click on the actions of the fragment and select "View usages"
* Propagate the changes to the pages
* Visit the page again

cache

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by jkappler** [comment]
ci:test:sf

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:test:echo

**Comment by jkappler** [comment]
@mbowerman, @dacousalr, fyi

File: modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java [hunk 1]
```diff
@@ -352,10 +351,7 @@ private String _renderFragmentEntryLink(
 		cacheKeySB.append(StringPool.DASH);
 		cacheKeySB.append(fragmentRendererContext.getLocale());
 		cacheKeySB.append(StringPool.DASH);
-		cacheKeySB.append(
-			StringUtil.merge(
-				fragmentRendererContext.getSegmentsEntryIds(),
-				StringPool.SEMICOLON));
+		cacheKeySB.append(fragmentEntryLink.getSegmentsExperienceId());
```
Comment (cgoncas) [inline modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java:354]: :)

**Comment by jkappler** [comment]
@david-gutierrez-mesa, it would be also good to have poshi tests for this, just in case

**Comment by david-gutierrez-mesa** [comment]
Task for Poshi automation created. See https://issues.liferay.com/browse/LPS-166299

**Review by lfbesada — APPROVED** [review]
Amazing as always ❤️

**Comment by lfbesada** [comment]
ci:forward

**Comment by jkappler** [comment]
Thank you @lfbesada ❤️! About the tests, I think it will fail only the first time because the user hasn't logged in 😢, thanks for fixing it

**Comment by jkappler** [comment]
ci:test:echo

**Comment by jkappler** [comment]
ci:test:tango

**Comment by ealonso** [comment]
ci:forward

**Pull request closed without merging**